### ISSUE TITLE
undefined reference to dpp::run

### ISSUE THREAD

**Issue by yuzarx** [luccanunes/discord.cpp#28]
Hi I got this error after runing the example on linux:
```
in function `main':                                                               discordmain.cpp:(.text+0x74): undefined reference to `dpp::Client::run(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&)'                            clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```
that's the command I used:
```
g++ -std=c++2a discordmain.cpp -ldiscord_cpp -lixwebsocket --stdlib=libc++ -lz
```
I copied the usr libs & includes to the default usr lib & includes `/usr/lib/` & `/usr/include/` and still same error

**Comment by luccanunes** [comment]
Hi there! Thanks for opening this issue.

Could you give us some information about your folder structure?

If i'm getting it right, here's a possible fix:
> An “Undefined Reference” error occurs when we have a reference to object name (class, function, variable, etc.) in our program and the linker cannot find its definition when it tries to search for it in all the linked object files and libraries.

As the above statement says, this is a linking error, to add the path to the binary files using g++ you can use `-L`

**Comment by yuzarx** [comment]
I didn't get it, however this is my file sturcture:
```
discordmain.cpp
discord.cpp/
    CMakeLists.txt  
    LICENSE  README.md  
    build  
    icon  
    ideas.md  
    src
```

**Comment by luccanunes** [comment]
Did you already build discord.cpp into binary files?

**Comment by yuzarx** [comment]
No, I used to build lib so files on other libs only, should I compile discord.cpp to one file ? or multiple files ?

**Comment by luccanunes** [comment]
You can compile as you wish, however, you'll have to link against those compiled files, since, as i mentioned, you're getting a linking error.
I'll be releasing an option for downloading the pre-compiled files soon.
You can also try the header-only version, available in the `header-only` branch, no linking necessary if you're using that.

**Comment by yuzarx** [comment]
swicthed to header-only branch it compiles without errors but, invalid token exception is raised even I regenerated the token

**Comment by luccanunes** [comment]
Do you have the latest version of the repo? (You might wanna check the `Client.hpp` file, see if there's a code 4004 validation)

**Comment by yuzarx** [comment]
```cpp
else if (msg->type == ix::WebSocketMessageType::Close) {
     switch (msg->closeInfo.code) {
          case 4004:                                                                                            
             isTokenInvalid = true;
             break;
     }
}
        
```

**Comment by luccanunes** [comment]
Ok, I'll look into it

**Comment by luccanunes** [comment]
Hi there, I keep trying but I still can't get the same thing to happen to me. Are you still using the example code?

**Comment by yuzarx** [comment]
I even re cloned the header branch, tried the same example but still getting invalid token exception, the same token works file with discord.py
